src/auto_scheduler/transform_step.cc

```diff
@@ -96,7 +96,12 @@ Step StepReadFromRecord(dmlc::JSONReader* reader) {
     return CacheReadStep(reader);
   } else if (name == CacheWriteStepNode::record_prefix_str) {
     return CacheWriteStep(reader);
-  } else {
+  } else if (name == FollowSplitStepNode::record_prefix_str) {
```

Comment:
Push this right behind the SplitStepNode. Also be care of the order in other positions.